### PR TITLE
Add a configuration generator

### PR DESCRIPTION
**Pull request by ChristianKaltenecker**
Adds a new tool called `config-generator`.
This tool exposes the `SampleSetParser` and `Solver` for either importing a sample set or generating all configurations for a given feature model.
This PR addresses se-sic/VaRA#838 (however, sampling strategies are missing; maybe open another Issue only for incorporating sampling strategies).

### CODE REVIEW

File: tools/config-generator/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  LINK_PRIVATE
+  VaRAFeature
+  VaRASampling
+  VaRASolver
+  csv
+  ${STD_FS_LIB}
+)
+
+add_custom_target(
+  check-vara-config-generator
```
Comment (vulder) [inline tools/config-generator/CMakeLists.txt:16]: You still need to add this test target to the github CI workflow

File: tools/config-generator/ConfigurationGenerator.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        llvm::cl::cat(ConfigCreatorCategory));
+
+static llvm::cl::opt<std::string>
+    CsvInputFilePath("csv", llvm::cl::desc("Path to the csv input file."),
+                     llvm::cl::value_desc("filename"), llvm::cl::init(""),
+                     llvm::cl::cat(ConfigCreatorCategory));
+
+static llvm::cl::opt<std::string>
+    OutputFilePath("out", llvm::cl::desc("Path to the yml output file."),
+                   llvm::cl::value_desc("filename"), llvm::cl::init(""),
```
Comment (vulder) [inline tools/config-generator/ConfigurationGenerator.cpp]:
```suggestion
                   llvm::cl::value_desc("filename"), llvm::cl::init("configurations.yaml"),
```

File: tools/config-generator/ConfigurationGenerator.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Errc.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/MemoryBuffer.h"
+
+static llvm::cl::OptionCategory
+    ConfigCreatorCategory("Configuration generator options");
+
+static llvm::cl::list<std::string>
+    FileNames(llvm::cl::Positional, llvm::cl::desc("Path to the feature model"),
```
Comment (vulder) [inline tools/config-generator/ConfigurationGenerator.cpp]: Why allow multiple file names if we only process one? We could just change this to a singe filename or?

File: tools/config-generator/ConfigurationGenerator.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    llvm::errs() << "error: Sampling is not implemented yet.\n";
+    return 1;
+  }
+
+  if (!OutputFilePath.empty() && !Configurations.empty()) {
+    const std::string Str =
+        vara::sampling::SampleSetWriter::writeConfigurations(*FM,
+                                                             Configurations);
+    std::error_code EC;
+    auto *Out = new llvm::raw_fd_ostream(OutputFilePath.getValue(), EC);
```
Comment (vulder) [inline tools/config-generator/ConfigurationGenerator.cpp]: Why not just use a value here? It's not really necessary to heap allocate the ostream here.

**Review by vulder — CHANGES_REQUESTED** [review]
A few nits but overall this looks good. Could you please also link the issue in the PR desc. so I know which issue to close :)

File: tools/config-generator/ConfigurationGenerator.cpp (hunk 3 above)
Comment (ChristianKaltenecker) [reply tools/config-generator/ConfigurationGenerator.cpp]: Yep. Changed it.

File: tools/config-generator/ConfigurationGenerator.cpp (hunk 4 above)
Comment (ChristianKaltenecker) [reply tools/config-generator/ConfigurationGenerator.cpp]: True; I changed this.

**Review by vulder — APPROVED** [review]

**Pull request merged**